pkg/sources/adapter/azureeventhubsource/adapter.go

```diff
@@ -146,28 +149,50 @@ func NewAdapter(ctx context.Context, envAcc pkgadapter.EnvConfigAccessor, ceClie
 
 // Start implements adapter.Adapter.
 func (a *adapter) Start(ctx context.Context) error {
+	go health.Start(ctx)
+
 	connCtx, cancel := context.WithTimeout(ctx, connTimeout)
 	runtimeInfo, err := a.ehClient.GetRuntimeInformation(connCtx)
 	cancel()
 	if err != nil {
-		a.logger.Fatal("GetRuntimeInformation failed:", err)
+		return fmt.Errorf("getting Event Hub runtime information: %w", err)
 	}
 	a.runtimeInfo = runtimeInfo
 
+	a.logger.Info("Starting Event Hub message receivers for partitions ", runtimeInfo.PartitionIDs)
+
 	// listen to each partition of the Event Hub
 	for _, partitionID := range runtimeInfo.PartitionIDs {
 		connCtx, cancel := context.WithTimeout(ctx, connTimeout)
 		_, err := a.ehClient.Receive(connCtx, partitionID, a.handleMessage, eventhub.ReceiveWithLatestOffset())
 		cancel()
 		if err != nil {
-			a.logger.Error("failed to start Event Hub listener:", err)
-			continue
+			a.logger.Errorw("An error occurred while starting message receivers. "+
```

Comment:
isnt it weird to put the wording `An error` in an error message? isn't that inferred?

Reply:
That review would be more useful with a suggestion to reword the message.

Reply:
Based on your other suggestion I think I understand what you mean now.

It's true that it is redundant to say it's an error when returning an error type.
However, this is not what we're doing here. We are logging that something happened in a particular partition, and the error that's actually returned below is more general about the fact that the adapter failed to start for X reason.

We are communicating in real time here something like this:
"oops something bad happened, we are now going to gracefully shut down and exit".
And because the shut down could take a while, we want the user to be notified as early as the problem occured.
Like I said, we are _also_ returning the error afterwards, but with a delay due to the shut down.